src/System.Net.Sockets/tests/FunctionalTests/SendFile.cs

```diff
@@ -170,86 +170,59 @@ public void SendFile_Synchronous(IPAddress listenAt, bool sendPreAndPostBuffers,
         [MemberData(nameof(SendFile_MemberData))]
         public void SendFile_APM(IPAddress listenAt, bool sendPreAndPostBuffers, int bytesToSend)
         {
-            const int ListenBacklog = 1;
-            const int LingerTime = 10;
-            const int TestTimeout = 30000;
+            const int ListenBacklog = 1, LingerTime = 10, TestTimeout = 30000;
 
             // Create file to send
-            byte[] preBuffer;
-            byte[] postBuffer;
+            byte[] preBuffer, postBuffer;
             Fletcher32 sentChecksum;
             string filename = CreateFileToSend(bytesToSend, sendPreAndPostBuffers, out preBuffer, out postBuffer, out sentChecksum);
 
             // Start server
-            var server = new Socket(listenAt.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
-            server.BindToAnonymousPort(listenAt);
-
-            server.Listen(ListenBacklog);
-
-            var serverFinished = new TaskCompletionSource<bool>();
-            int bytesReceived = 0;
-            var receivedChecksum = new Fletcher32();
-
-            server.AcceptAPM(remote =>
+            using (var listener = new Socket(listenAt.AddressFamily, SocketType.Stream, ProtocolType.Tcp))
             {
-                Action<int> recvHandler = null;
-                bool first = true;
+                listener.BindToAnonymousPort(listenAt);
+                listener.Listen(ListenBacklog);
 
-                var recvBuffer = new byte[256];
-                recvHandler = received => 
+                int bytesReceived = 0;
+                var receivedChecksum = new Fletcher32();
+
+                Task serverTask = Task.Run(async () =>
                 {
-                    if (!first)
+                    using (var serverStream = new NetworkStream(await listener.AcceptAsync(), ownsSocket: true))
                     {
-                        if (received == 0)
+                        var buffer = new byte[256];
+                        int bytesRead;
+                        while ((bytesRead = await serverStream.ReadAsync(buffer, 0, buffer.Length)) != 0)
```

Comment:
Because this code is more concise and the BeginReceive was part of the test rather than what was being tested.